lib/benchee/formatters/json.ex

```diff
@@ -49,57 +48,74 @@ defmodule Benchee.Formatters.JSON do
   JSON with run_times and statistics.
 
   ## Examples
-
-      iex> suite = %{
-      ...>   run_times: %{"Some Input" =>
-      ...>     %{"My Job" => [200, 400, 400, 400, 500, 500, 700, 900]}},
-      ...>   statistics: %{"Some Input" =>
-      ...>     %{"My Job" => %{average: 500.0, ips: 2000.0, std_dev: 200.0,
-      ...>       std_dev_ratio: 0.4, std_dev_ips: 800.0, median: 450.0}}}}
+      iex> suite = %Benchee.Suite{
+      ...>      scenarios: [
+      ...>        %Benchee.Benchmark.Scenario{
+      ...>          job_name: "My Job",
+      ...>          run_times: [200, 400, 400, 400, 500, 500, 700, 900],
+      ...>          input_name: "Some Input",
+      ...>          input: "Some Input",
+      ...>          run_time_statistics: %Benchee.Statistics{
+      ...>            average:       500.0,
+      ...>            ips:           2000.0,
+      ...>            std_dev:       200.0,
+      ...>            std_dev_ratio: 0.4,
+      ...>            std_dev_ips:   800.0,
+      ...>            median:        450.0,
+      ...>            minimum:       200,
+      ...>            maximum:       900,
+      ...>            sample_size:   8
+      ...>          }
+      ...>        },
+      ...>      ]
+      ...>    }
       iex> Benchee.Formatters.JSON.format(suite)
-      %{
-        "Some Input" =>
-          "{\\"statistics\\":{\\"My Job\\":{\\"std_dev_ratio\\":0.4,\\"std_dev_ips\\":800.0,\\"std_dev\\":200.0,\\"median\\":450.0,\\"ips\\":2.0e3,\\"average\\":500.0}},\\"sort_order\\":[\\"My Job\\"],\\"run_times\\":{\\"My Job\\":[200,400,400,400,500,500,700,900]}}"
-      }
+      %{"Some Input" => "{\\"statistics\\":{\\"My Job\\":{\\"std_dev_ratio\\":0.4,\\"std_dev_ips\\":800.0,\\"std_dev\\":200.0,\\"sample_size\\":8,\\"minimum\\":200,\\"median\\":450.0,\\"maximum\\":900,\\"ips\\":2.0e3,\\"average\\":500.0}},\\"sort_order\\":[\\"My Job\\"],\\"run_times\\":{\\"My Job\\":[200,400,400,400,500,500,700,900]}}"}
 
   """
-  @spec format(Benchee.Suite.t) :: %{Benchee.Suite.key => String.t}
-  def format(%{statistics: statistics, run_times: run_times}) do
-    statistics
-    |> Enum.map(fn({input, statistics_map}) ->
-         run_times_list = Map.fetch!(run_times, input)
-         {input, format_measurements(statistics_map, run_times_list)}
+
+  @spec format(Suite.t) :: %{Suite.key => String.t}
+  def format(%Suite{scenarios: scenarios}) do
+    scenarios
+    |> Enum.group_by(fn(scenario) -> scenario.input_name end)
+    |> Enum.map(fn({input, scenarios}) ->
+         {input, format_scenarios_for_input(scenarios)}
        end)
     |> Map.new
```

Comment:
(speaking about gropuing, adding statistics etc.. just the sort order which we might need, but that's only relevant within one input... so it makes it more interesting. We might be able to live without it though? Not sure, question for the html formatter)